controllers/dscinitialization/suite_test.go

```diff
@@ -62,18 +92,63 @@ var _ = BeforeSuite(func() {
 	Expect(err).NotTo(HaveOccurred())
 	Expect(cfg).NotTo(BeNil())
 
-	err = dscinitializationv1.AddToScheme(scheme.Scheme)
-	Expect(err).NotTo(HaveOccurred())
-
+	utilruntime.Must(clientgoscheme.AddToScheme(scheme.Scheme))
+	utilruntime.Must(dscinitializationv1.AddToScheme(scheme.Scheme))
+	utilruntime.Must(netv1.AddToScheme(scheme.Scheme))
+	utilruntime.Must(authv1.AddToScheme(scheme.Scheme))
+	utilruntime.Must(corev1.AddToScheme(scheme.Scheme))
+	utilruntime.Must(apiextv1.AddToScheme(scheme.Scheme))
+	utilruntime.Must(routev1.AddToScheme(scheme.Scheme))
+	utilruntime.Must(appsv1.AddToScheme(scheme.Scheme))
+	utilruntime.Must(ofapi.AddToScheme(scheme.Scheme))
 	//+kubebuilder:scaffold:scheme
 
+	// Initialize Kubernetes client
 	k8sClient, err = client.New(cfg, client.Options{Scheme: scheme.Scheme})
 	Expect(err).NotTo(HaveOccurred())
 	Expect(k8sClient).NotTo(BeNil())
 
+	// Setup controller manager
+	mgr, err := ctrl.NewManager(cfg, ctrl.Options{
+		Scheme:             scheme.Scheme,
+		LeaderElection:     false,
+		MetricsBindAddress: "0",
+	})
+
+	Expect(err).NotTo(HaveOccurred())
+
+	err = (&DSCInitializationReconciler{
+		Client:                k8sClient,
+		Scheme:                scheme.Scheme,
+		Log:                   ctrl.Log.WithName("controllers").WithName("DSCInitialization"),
+		Recorder:              mgr.GetEventRecorderFor("dscinitialization-controller"),
+		ApplicationsNamespace: AppNamespace,
+	}).SetupWithManager(mgr)
+
+	Expect(err).ToNot(HaveOccurred())
+
+	// Start the manager
+	go func() {
+		defer GinkgoRecover()
+		err = mgr.Start(ctx)
+		Expect(err).ToNot(HaveOccurred(), "Failed to run manager")
+	}()
+})
+
+// Cleanup resources to not contaminate between tests
+var _ = AfterEach(func() {
+	inNamespace := client.InNamespace(WorkingNamespace)
+	appNamespace := client.InNamespace(AppNamespace)
+	Expect(k8sClient.DeleteAllOf(context.TODO(), &dscinitializationv1.DSCInitialization{}, inNamespace)).ToNot(HaveOccurred())
+	Expect(k8sClient.DeleteAllOf(context.TODO(), &netv1.NetworkPolicy{}, appNamespace)).ToNot(HaveOccurred())
+	Expect(k8sClient.DeleteAllOf(context.TODO(), &corev1.ConfigMap{}, appNamespace)).ToNot(HaveOccurred())
+	Expect(k8sClient.DeleteAllOf(context.TODO(), &authv1.RoleBinding{}, appNamespace)).ToNot(HaveOccurred())
```

Comment:
I would expect those resources to be deleted when the owning resource is deleted, so I'm not quite sure this should belong to clean-up logic. Perhaps only the created `DSCI`, but then I would probably put it to the test itself, by using `defer`, so the intent is immediately visible to the reader.

Reply:
I guess manual cleanup is required : https://book.kubebuilder.io/reference/envtest.html#testing-considerations